torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -0,0 +1,179 @@
+#include <torch/csrc/jit/backends/backend_init.h>
+#include <torch/csrc/jit/backends/backend.h>
+#include <torch/csrc/jit/runtime/custom_operator.h>
+#include "torch/csrc/jit/frontend/code_template.h"
+
+namespace torch {
+namespace jit {
+void initJitBackendBindings(PyObject* module) {
+  // Prefixes for looking up preprocessor and creator methods.
+  const c10::QualifiedName backend_prefix("torch.__backends__");
+  const c10::QualifiedName backend_creator_prefix(
+      backend_prefix, "__creator__");
+  const c10::QualifiedName backend_preprocess_prefix(
+      backend_prefix, "__preprocess__");
+
+  // For each backend:
+  for (auto& fn : getCustomClassMethodsWithPrefix(backend_creator_prefix)) {
+    auto backend_name = fn->name();
+    auto backend_creator_name = fn->qualname();
+
+    // Create a method named to_<backend> and pybind it.
+    auto m = py::handle(module).cast<py::module>();
+    std::string to_backend_method_name = "_jit_to_" + backend_name;
+    m.def(
+        to_backend_method_name.c_str(),
+        [=](Module orig_module, py::dict method_compile_spec) {
+          // TODO: Validate method_compile_spec.
+
+          // Clone orig_module to make sure backend transformation is
+          // functional.
+          auto cloned_module = orig_module.clone();
+
+          // Represents of a Type of Dict[str, Any].
+          auto any_dict_ty =
+              DictType::create(StringType::get(), AnyType::get());
+
+          // Call preprocess.
+          auto* preprocess_fn = getCustomClassMethod(
+              c10::QualifiedName(backend_preprocess_prefix, backend_name));
+          TORCH_CHECK(
+              preprocess_fn,
+              "Backend ",
+              backend_name,
+              " was not found; did you forget to register it?");
+          Stack stack;
+          push(
+              stack,
+              cloned_module._ivalue(),
+              toIValue(method_compile_spec, any_dict_ty).toGenericDict());
+          preprocess_fn->run(stack);
+          auto preprocessed_module = pop(stack);
```

Comment:
Do we expect the stack to be empty after this line? Should probably add an assert.

Alternatively above you could use `Function::operator()`, which takes a stack and returns the single IValue return value rather than relying on Stack pre/postconditions